### PR TITLE
[DOC-392] Local install with Proxy and infra recommendations

### PR DESCRIPTION
**Pull request by nadegepepin**

### CODE REVIEW

File: doc/docs/master/deploy-manage/deploy/azure.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ For a quick test installation of Pachyderm on Azure (suitable for development),
       - Read our [infrastructure recommendations](../ingress/). You will find instructions on how to set up an ingress controller, a load balancer, or connect an Identity Provider for access control. 
       - If you are planning to install Pachyderm UI. Read our [Console deployment](../console/) instructions. Note that, unless your deployment is `LOCAL` (i.e., on a local machine for development only, for example, on Minikube or Docker Desktop), the deployment of Console requires, at a minimum, the set up on an Ingress.
 
+!!! Attention 
+    We are now shipping Pachyderm with an **optional embedded proxy ([Envoy](https://www.envoyproxy.io/))** 
+    allowing your cluster to expose one single port to the Internet.
```
Comment (tybritten) [inline doc/docs/master/deploy-manage/deploy/azure.md:13]:
```suggestion
    allowing your cluster to expose one single port externally.
```

File: doc/docs/master/deploy-manage/deploy/aws-deploy-pachyderm.md [hunk 2]
```diff
@@ -7,6 +7,12 @@ For a quick test installation of Pachyderm on AWS (suitable for development), ju
       - Read our [infrastructure recommendations](../ingress/){target=_blank}. You will find instructions on how to set up an ingress controller, a load balancer, or connect an Identity Provider for access control. 
       - If you are planning to install Pachyderm UI. Read our [Console deployment](../console/){target=_blank} instructions. Note that, unless your deployment is `LOCAL` (i.e., on a local machine for development only, for example, on Minikube or Docker Desktop), the deployment of Console requires, at a minimum, the set up on an [Ingress](../ingress/#ingress).
 
+!!! Attention 
+    We are now shipping Pachyderm with an **optional embedded proxy ([Envoy](https://www.envoyproxy.io/))**
```
Comment (jrockway) [inline doc/docs/master/deploy-manage/deploy/aws-deploy-pachyderm.md:11]: I kind of wanted to keep the identity of the proxy somewhat hidden.  Users can't configure it in any way, and I'd like the option to switch to something else without anyone noticing.

File: doc/docs/master/deploy-manage/deploy/envoy.md [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+        annotations: {see examples below}
+    ```
+
+
+    === "Example on AWS EKS"
+        In the following example, we deploy an NLB on AWS EKS:
```
Comment (jrockway) [inline doc/docs/master/deploy-manage/deploy/envoy.md:51]: Why are we choosing an NLB here?  I find that they work very poorly with k8s; a "classic" load balancer is fine though.

File: doc/docs/master/deploy-manage/deploy/envoy.md [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+    When Envoy is enabled with `type:LoadBalancer` (see the snippet of values.yaml enabling the proxy), Pachyderm creates a `pachyderm-proxy` service allowing your cloud platform (AWS, GKE...) to **provision a TCP Load Balancer automatically**.
+        
+    To provision this external load balancer automatically (if supported) and attach any Load Balancer configuration information to the metadata of your servic, add the appropriate `annotations` in the `proxy.service` of your values.yaml (see below):
+
+    ```yaml
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+        annotations: {see examples below}
```
Comment (jrockway) [inline doc/docs/master/deploy-manage/deploy/envoy.md:46]: I think the case where annotations are needed is pretty rare.  The default load balancer from cloud providers is the right one most of the time.

File: doc/docs/master/deploy-manage/deploy/envoy.md [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+
+    === "Example on GCP GKE"
+        In the following example, we pre-created a static IP by running `gcloud compute addresses create ADDRESS_NAME --global --ip-version IPV4`, then passed this external IP to the values.yaml as follow:
+
+        ``` yaml
+        proxy:
+          enabled: true
+          service:
+            type: LoadBalancer
+            loadBalancerIP: ${ADDRESS_NAME}
```
Comment (jrockway) [inline doc/docs/master/deploy-manage/deploy/envoy.md:74]: Does this name based binding work?  I assume you have to do something like:

```
$ gcloud compute addresses create jonathan-test-address-deleteme
Created [https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1/addresses/jonathan-test-address-deleteme].

$ gcloud compute addresses describe jonathan-test-address-deleteme
address: 34.67.7.54
addressType: EXTERNAL
creationTimestamp: '2022-06-03T14:07:02.870-07:00'
description: ''
id: '3059108090574522649'
kind: compute#address
name: jonathan-test-address-deleteme
networkTier: PREMIUM
region: https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1
selfLink: https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1/addresses/jonathan-test-address-deleteme
status: RESERVED
```

And then put `34.67.7.54` in the loadBalancerIP field.

File: doc/docs/master/deploy-manage/deploy/envoy.md [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+
+        ``` yaml
+          proxy:
+            enabled: true
+            service:
+              type: LoadBalancer
+              loadBalancerIP: ${ADDRESS_NAME}
+        ```
+
+* **Use a secure connection**
```
Comment (jrockway) [inline doc/docs/master/deploy-manage/deploy/envoy.md:87]: This won't work with the proxy.  pachd and console TLS has to be enabled, and then the secret has to be in `envoy-tls`, and `proxy.tls.enabled` in the values will need to be true.  (This code isn't merged yet, however.)

**Review by jrockway — APPROVED** [review]
This looks very good.  Github won't let me comment on images, but the diagram has that purple line unnecessarily crossing the blue one again ;)

File: doc/docs/master/deploy-manage/deploy/azure.md (hunk 1 above)
Comment (nadegepepin) [reply doc/docs/master/deploy-manage/deploy/azure.md]: Done

File: doc/docs/master/deploy-manage/deploy/aws-deploy-pachyderm.md (hunk 2 above)
Comment (nadegepepin) [reply doc/docs/master/deploy-manage/deploy/aws-deploy-pachyderm.md]: Done - Used the generic term of Proxy

File: doc/docs/master/deploy-manage/deploy/envoy.md (hunk 3 above)
Comment (nadegepepin) [reply doc/docs/master/deploy-manage/deploy/envoy.md]: removed all annotations

File: doc/docs/master/deploy-manage/deploy/envoy.md (hunk 4 above)
Comment (nadegepepin) [reply doc/docs/master/deploy-manage/deploy/envoy.md]: removed all annotations

File: doc/docs/master/deploy-manage/deploy/envoy.md (hunk 6 above)
Comment (nadegepepin) [reply doc/docs/master/deploy-manage/deploy/envoy.md]: I removed the TLS section for now with a Coming soon mention.

**Pull request merged**